nav2_toolkit/include/nav2_toolkit/pose_saver_node.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef POSE_SAVER__POSE_SAVER_NODE_HPP_
+#define POSE_SAVER__POSE_SAVER_NODE_HPP_
+
+#include <rclcpp/rclcpp.hpp>
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <std_srvs/srv/trigger.hpp>
+#include <yaml-cpp/yaml.h>
+
+namespace nav2_toolkit
+{
+
+class PoseSaverNode : public rclcpp::Node
```

Comment:
Class and metrics need doxygen documentation. See other packages for reference :-)

Reply:
Also remove any commented out values that are unused.